### PR TITLE
chore: make `[Display|Icon|Window]Id` more linker friendly

### PR DESCRIPTION
**Pull request by spouliot**
Allow operator methods to be removed when `Equals` is used.

GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

- Refactoring (no functional changes, no api changes)

## What is the current behavior?

The linker has to include more code than what's really required.

## What is the new behavior?

`Equals(object)` is an override from `System.Object` so it will always be marked. `Equals(T)` is from `IEquatable<T>` and almost aways marked. 
Others should be trimmable by using the non-trimmable parts.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWP/Microsoft/UI/DisplayId.cs [hunk 1]
```diff
@@ -13,20 +13,13 @@ public DisplayId(ulong _Value)
...
 		public override bool Equals(object obj)
-		{
-			if (obj is DisplayId)
-			{
-				DisplayId y = (DisplayId)obj;
-				return this == y;
-			}
-			return false;
-		}
+			=> (obj is DisplayId y) && Equals(y);
```
Comment (Youssef1313) [inline src/Uno.UWP/Microsoft/UI/DisplayId.cs:22]: I think the parentheses are unnecessary.

```suggestion
			=> obj is DisplayId y && Equals(y);
```

File: src/Uno.UWP/Microsoft/UI/IconId.cs [hunk 2]
```diff
@@ -17,20 +17,13 @@ public IconId(ulong _Value)
...
 		public override bool Equals(object obj)
-		{
-			if (obj is IconId)
-			{
-				IconId y = (IconId)obj;
-				return this == y;
-			}
-			return false;
-		}
+			=> (obj is IconId y) && Equals(y);
```
Comment (Youssef1313) [inline src/Uno.UWP/Microsoft/UI/IconId.cs:26]:
```suggestion
			=> obj is IconId y && Equals(y);
```

File: src/Uno.UWP/Microsoft/UI/WindowId.cs [hunk 3]
```diff
@@ -13,20 +13,13 @@ public WindowId(ulong _Value)
...
 		public override bool Equals(object obj)
-		{
-			if (obj is WindowId)
-			{
-				WindowId y = (WindowId)obj;
-				return this == y;
-			}
-			return false;
-		}
+			=> (obj is WindowId y) && Equals(y);
```
Comment (Youssef1313) [inline src/Uno.UWP/Microsoft/UI/WindowId.cs:22]:
```suggestion
			=> obj is WindowId y && Equals(y);
```

**Review by Youssef1313 — APPROVED** [review]

**Review by jeromelaban — APPROVED** [review]

**Comment by jeromelaban** [comment]
Ah my apologies, I was a bit too fast on merging (the android issue was blocking the merge for no good reason, but there were youssef's comments).

**Comment by Youssef1313** [comment]
No problem, the comments aren't too important any way :)

**Pull request merged**